stub/src/test/java/io/grpc/stub/ClientCallsTest.java

```diff
@@ -99,6 +99,124 @@ public void tearDown() {
     }
   }
 
+  @Test
+  public void unaryAsyncCallStatusIsOkWithMessageSuccess() throws Exception {
+    Integer req = 2;
+    final String resp = "bar";
+    final Status status = Status.OK;
+    final Metadata trailers = new Metadata();
+    final List<String> actualResponse = new ArrayList<>();
+    final List<Boolean> completed = new ArrayList<>();
+
+    NoopClientCall<Integer, String> call = new NoopClientCall<Integer, String>() {
+      @Override
+      public void start(ClientCall.Listener<String> listener, Metadata headers) {
+        listener.onMessage(resp);
+        listener.onClose(status, trailers);
+      }
+    };
+
+    StreamObserver<String> responseObserver = new StreamObserver<String>() {
+      @Override
+      public void onNext(String value) {
+        actualResponse.add(value);
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        fail("Should not reach here");
+      }
+
+      @Override
+      public void onCompleted() {
+        completed.add(true);
+      }
+    };
+
+    ClientCalls.asyncUnaryCall(call, req, responseObserver);
+    assertThat(actualResponse.size()).isEqualTo(1);
+    assertEquals(resp, actualResponse.get(0));
+    assertThat(completed.size()).isEqualTo(1);
+    assertThat(completed.get(0)).isTrue();
+  }
+
+  @Test
+  public void unaryAsyncCallStatusIsOkWithNullMessageGetError() throws Exception {
+    Integer req = 2;
+    final Status status = Status.OK;
+    final Metadata trailers = new Metadata();
+    final List<Throwable> expected = new ArrayList<>();
+
+    NoopClientCall<Integer, String> call = new NoopClientCall<Integer, String>() {
+      @Override
+      public void start(ClientCall.Listener<String> listener, Metadata headers) {
+        listener.onMessage(null);
+        listener.onClose(status, trailers);
+      }
+    };
+
+    StreamObserver<String> responseObserver = new StreamObserver<String>() {
+      @Override
+      public void onNext(String value) {
+        fail("Should not reach here");
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        expected.add(t);
+      }
+
+      @Override
+      public void onCompleted() {
+        fail("Should not reach here");
+      }
+    };
+
+    ClientCalls.asyncUnaryCall(call, req, responseObserver);
+    assertThat(expected.size()).isEqualTo(1);
+    assertThat(expected.get(0).getMessage())
```

Comment:
Use `assertThat(expected.get(0)).hasMessageThat()` to produce a better error message when it fails. Ditto below.